File: P1S2/ruby/grupo_teoria.rb
require_relative 'grupo'

class GrupoTeoria < Grupo
  def initialize(nombre)
    super
    puts "Creado grupo de teoría #{@nombre}"
  end

  def to_s
    "GrupoTeoria{nombre='#{@nombre}', aula=#{@aula&.nombre}}"
  end
end

File: P1S2/P1S2Java/src/GrupoTeoria.java
public class GrupoTeoria extends Grupo {
    private AulaTeoria aula;

    public GrupoTeoria(String nombre) {
        super(nombre);
        System.out.println("teoría.");
    }

    @Override
    void asignarAula(Aula aula) {
        this.aula = (AulaTeoria) aula;
    }

    @Override
    public String toString() {
        return "GrupoTeoria{" +
                "nombre='" + nombre + '\'' +
                ", aula=" + aula +
                '}';
    }
}
